go/src/meguca/db/reader.go

```diff
@@ -246,6 +249,8 @@ func extractPost(ps postScanner, is imageScanner) (p common.Post, err error) {
 	if p.Image != nil {
 		p.Image.Spoiler, p.Image.Name = ps.Image()
 	}
+
+	err = getPostModLog(&p)
```

Comment:
Please only perform, if post is banned, meidovisioned, etc.

Reply:
`getPostModLog` only populates if any of those are true anyway, no point in adding a check outside of it, it will just return err and run a loop 3 and (in this case, if none are true) do nothing and return.

Reply:
But it will also start a transaction for no reason.

Reply:
Are transactions heavy enough to matter?
If so, I'll change it.

Reply:
Please remember this for the rest of your life: All I/O is heavy enough to matter.

Reply:
Oh, so it's doing I/O anyway.
I just didn't know whether it was worth the extra check.